src/line-number-gutter-component.coffee

```diff
@@ -1,13 +1,13 @@
-_ = require 'underscore-plus'
-{setDimensionsAndBackground} = require './gutter-component-helpers'
-
+TiledComponent = require './tiled-component'
+LineNumbersTileComponent = require './line-numbers-tile-component'
 WrapperDiv = document.createElement('div')
+DummyLineNumberComponent = LineNumbersTileComponent.createDummy()
 
 module.exports =
-class LineNumberGutterComponent
+class LineNumberGutterComponent extends TiledComponent
   dummyLineNumberNode: null
 
-  constructor: ({@onMouseDown, @editor, @gutter}) ->
+  constructor: ({@onMouseDown, @editor, @presenter, @gutter}) ->
```

Comment:
Not really, as we are already passing it in `LinesComponent` as well as in `LinesTileComponent`. I agree, though, this is an unhealthy dependency and, as far as I remember, we use it only for testing purposes (e.g. `lineNodeForScreenRow` and `lineNumberNodeForScreenRow`).

I am going to get rid of it everywhere, exception made for `TextEditorComponent` which should be the only object to know about it. The correspondence between tiles and rows will still be handled by `TextEditorPresenter` but its conversion methods will be called in one place only (i.e. `TextEditorComponent`). I'll /cc you in the relevant upcoming commit.

Nice suggestion! :sparkler: